src/backend/distributed/commands/alter_table.c

```diff
@@ -1365,7 +1368,16 @@ CreateCitusTableLike(TableConversionState *con)
 {
 	if (IsCitusTableType(con->relationId, DISTRIBUTED_TABLE))
 	{
-		CreateDistributedTableLike(con);
+		if (IsCitusTableType(con->relationId, SINGLE_SHARD_DISTRIBUTED))
+		{
+			char *newColocateWith =
+				quote_qualified_identifier(con->schemaName, con->relationName);
+			CreateSingleShardTable(con->newRelationId, newColocateWith);
```

Comment:
I like using "single shard table" in the code, for readability purposes. Separate function for single shard tables seems more clean to me. But I'm open for a discussion about that.

Reply:
I feel like we usually want to treat single shard tables as a variant of distributed tables, where the shard key is null and the shard count is strictly equal to one. This is also true for most pieces of the code-base and a good practice from the re-usability perspective.

Even more, the callers wouldn't need to think about whether to call CreateDistributedTableLike or  CreateSingleShardTableLike, if we add this logic into CreateDistributedTableLike instead of defining a separate function.